src/libmongoc/tests/unified/runner.c

```diff
@@ -716,6 +716,33 @@ check_run_on_requirement (test_runner_t *test_runner,
          continue;
       }
 
+      if (0 == strcmp (key, "serverless")) {
+         const char *serverless_mode = bson_iter_utf8 (&req_iter, NULL);
+
+         if (0 == strcmp (serverless_mode, "allow")) {
+            return true;
```

Comment:
This return early may skip additional run on requirements to check.
```suggestion
            continue;
```

Reply:
I can't resist ([PHPLIB-680](https://jira.mongodb.org/browse/PHPLIB-680)).

![](https://media.giphy.com/media/HTkQXTBHzAjpS/giphy.gif)

Reply:
I blame years of writing early exit clauses due to https://github.com/slevomat/coding-standard#slevomatcodingstandardcontrolstructuresearlyexit-

On a separate note, I had half a mind of extracting checks to static functions to ensure such problems no longer happen. I'll do that in a separate PR as I've noticed it in several places.